src/OpenRiaServices.EntityFramework/Framework/DbDomainService.cs

```diff
@@ -304,5 +304,68 @@ private static async Task SetChangeSetConflictsAsync(Dictionary<DbEntityEntry, C
                 }
             }
         }
+
+        /// <summary>
+        /// Mark the entity as added
+        /// </summary>
+        /// <typeparam name="TEntity">Entity type</typeparam>
+        /// <param name="entity">The modified entity</param>
+        /// <param name="dbSet">The db set</param>
+        protected void AttachAsAdded<TEntity>(TEntity entity, DbSet<TEntity> dbSet = null)
+            where TEntity : class
+        {
+            EntityEntry<TEntity> entityEntry = DbContext.Entry(entity);
+
+            if (entityEntry.State != EntityState.Detached)
+            {
+                entityEntry.State = EntityState.Added;
+            }
+            else
+            {
+                dbSet ??= DbContext.Set<TEntity>();
+                dbSet.Add(entity);
```

Comment:
we consider skippting dbset parameter.

We either remove method from ef6 or use 
```
#if EFCORE
DbContext.Add
#else
DbContext.Set<TEntity>().Add(...);
#endif
```